duckari/duck.py

```diff
@@ -25,13 +26,25 @@ def decorate(func: _ListenerT) -> _ListenerT:
 
         return decorate
 
+    def command(self, name: str = None): # TODO: Add return type
+        def decorate(func: _ListenerT) -> _ListenerT:
+            self._commands[name].append(func) # You can add multiple callbacks for a command.
+            return func
+
+        return decorate
+
     async def on_message_create(self, event: hikari.MessageCreateEvent) -> None:
         assert self._prefix is not None
 
         if event.message.author.is_bot:
             return
 
-        # Do command processing here :beanos:
+        if event.message.content.startswith(self._prefix):
+            content_without_prefix = event.message.content.replace(self._prefix, "", 1)
+            if content_without_prefix.split()[0] in self._commands.keys():
+                for callback in self._commands[content_without_prefix.split()[0]]:
+                    callback() # TODO: Add context
```

Comment:
Generally you need to await asynchronous functions...

Reply:
Right, I forgot.